Rubberduck.Parsing/VBA/RubberduckParser.cs

```diff
@@ -161,19 +159,55 @@ private void ParseAll()
                 _componentAttributes.Remove(invalidated);
             }
 
-            foreach (var vbComponent in modified)
+            foreach (var vbComponent in toParse)
             {
                 ParseAsync(vbComponent, CancellationToken.None);
             }
         }
 
+        private void AddBuiltInDeclarations(IReadOnlyList<VBProject> projects)
+        {
+            SyncComReferences(projects);
+
+            var finder = new DeclarationFinder(_state.AllDeclarations, new CommentNode[]{}, new IAnnotation[]{});
+            if (finder.MatchName(Tokens.Err).Any(item => item.IsBuiltIn 
+                && item.DeclarationType == DeclarationType.Variable 
+                && item.Accessibility == Accessibility.Global))
+            {
+                return;
+            }
+            
+            var vba = finder.FindProject("VBA");
+            Debug.Assert(vba != null);
+            
+            var errObject = finder.FindClass(vba, "ErrObject", true);
+            Debug.Assert(errObject != null);
+
+            var qualifiedName = new QualifiedModuleName(vba.IdentifierName, vba.IdentifierName, errObject.IdentifierName);
+            var err = new Declaration(new QualifiedMemberName(qualifiedName, Tokens.Err), vba, "Global", errObject.IdentifierName, true, false, Accessibility.Global, DeclarationType.Variable);
+            _state.AddDeclaration(err);
+
+            var debugClassName = new QualifiedModuleName(vba.IdentifierName, vba.IdentifierName, "DebugClass");
```

Comment:
It's because the Project Ids don't match. The newly created QualifiedModuleName should use the same constructor as the one used for references.

The Err works just by chance, because the original ErrObject has a "reference" project id as QualifiedModuleName.

Here's a fix:

```
            var debugClassName = new QualifiedModuleName(vba.QualifiedName.QualifiedModuleName.ProjectName, vba.QualifiedName.QualifiedModuleName.ProjectPath, "DebugClass");
            var debugClass = new ClassModuleDeclaration(new QualifiedMemberName(debugClassName, "DebugClass"), vba, "DebugClass", true, new List<IAnnotation>(), new Attributes(), true);
```

ProjectName and ProjectPath need to be made available through Getter-Properties in the QualifiedModuleName-class. This works, just tested it.